bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/NvsBeanValidator.java

```diff
@@ -99,7 +100,7 @@ private String validateValue(String value, String type, String encoding)
 
 		if (encoding.contentEquals("string") || encoding.contentEquals("binary")) //$NON-NLS-1$ //$NON-NLS-2$
 		{
-			if (value.length() > 4000)
+			if (value.getBytes().length > STRING_VALUE_LIMIT)
```

Comment:
the bug was here because character size can be different for different encodings and we have a limitation of 4000 bytes